test/unit/processing/payment/get-latest-payment.test.js

```diff
@@ -0,0 +1,162 @@
+const getLatestPayment = require('../../../../app/processing/payment/get-latest-payment')
+let paymentRequest
+let settlement
+let lastSettlement
+let instalmentValue
```

Comment:
missing tests for if `paymentRequest.schedule` is `false`

Reply:
_alsssoooo_ I guess this where a clear definition of unit vs integration tests come in.

I have that a unit test should mock any functions it imports (and hopefully therefore calls) and define their return values.

So we only test in unit tests:

- that it calls (or doesn't) a function
- it calls it how many times
- it calls it w/ what params
- given that this function returns this, what would the unit tested function return
- given that this function returns that, what would the unit tested function return
- etc
- ofc if the function being unit tested doesn't "import" any functions then it only has return check tests, such as `getPeriodFromSchedule` would do (dispite it using `Math.trunc` but built-ins are different... _maybe_?)


In an ideal world, I would like a broken down unit test for this as described above (if tht is what our team wanted inna collaborative discussion... dev chat next topic anyone 👁️) but, this is a fairly simple func so idk

Reply:
Still missing is `paymentRequest.schedule` is `false`